### ISSUE TITLE
ChromeLauncher errors

### ISSUE THREAD

**Issue by ebidel** [GoogleChrome/lighthouse#2258]
`yarn add lighthouse@^2.x`

Then run this code:

```
const Lighthouse = require('lighthouse');
const ChromeLauncher = require('lighthouse/chrome-launcher/chrome-launcher');

function launchChromeAndRunLighthouse(url, flags, config = null) {
  return ChromeLauncher.launch()                         // Launch Chrome.
    .then(() => Lighthouse(url, flags, config))          // Run Lighthouse.
    .then(results => chrome.kill().then(() => results)); // Kill Chrome, results.
}

const flags = {output: 'json'};
launchChromeAndRunLighthouse('https://example.com', flags).then(results => {
  console.log(results);
});
```

Produces a error for a missing compiled file:

<img width="517" alt="screen shot 2017-05-15 at 7 11 47 pm" src="https://cloud.githubusercontent.com/assets/238208/26088000/26b27b6c-39a9-11e7-8c35-147564699ed6.png">

When I build that file, I still get errors:

<img width="532" alt="screen shot 2017-05-15 at 7 18 08 pm" src="https://cloud.githubusercontent.com/assets/238208/26087961/ee61cfd8-39a8-11e7-9e41-1e5e91fa1184.png">

**Comment by samccone** [comment]
investigating

**Comment by samccone** [comment]
from the root of lighthouse:

```js
const Lighthouse = require('./lighthouse-core/index.js');
const ChromeLauncher = require('./chrome-launcher/chrome-launcher.js');

function launchChromeAndRunLighthouse(url, flags, config = null) {
  let chrome;

  return ChromeLauncher.launch().then(v => {
    chrome = v;
    flags.port = v.port;
    return Lighthouse(url, flags, config);
  }).then(results => {
    return chrome.kill().then(() => results);
  });
}

const flags = {
  output: 'json'
};
launchChromeAndRunLighthouse('https://example.com', flags).then(results => {
  console.log(results);
}).catch(e => console.log(e));
```

--

The other issue was just a publish issue it looks like

**Comment by samccone** [comment]
another option if you want to maybe have port collisions 

![image](https://cloud.githubusercontent.com/assets/883126/26088912/aad61354-39ae-11e7-894a-36d6425f7231.png)

**Comment by samccone** [comment]
over to @paulirish for the publish fix / issue

**Comment by paulirish** [comment]
will fix the publish!

**Comment by samccone** [comment]
![image](https://cloud.githubusercontent.com/assets/883126/26089293/691f161a-39b1-11e7-910c-5e1854e3d87f.png)

**Comment by paulirish** [comment]
FTFY

```js
const Lighthouse = require('./lighthouse-core/index.js');
const chromeLauncher = require('./chrome-launcher/chrome-launcher.js');

function launchChromeAndRunLighthouse(url, flags, config = null) {
  return chromeLauncher.launch().then(chrome => {
    flags.port = chrome.port;
    return Lighthouse(url, flags, config).then(results =>
      chrome.kill().then(() => results)
    );
  });
}


// let's run this!
const flags = {output: 'json'};
launchChromeAndRunLighthouse('https://example.com', flags)
  .then(results => console.log(results));
```

**Comment by ebidel** [comment]
@paulirish that might be the best one for the readme. `async/await` are still out of reach for people. Do you want to PR it?

But use the packages the way consumers will:

```
const Lighthouse = require('lighthouse');
const chromeLauncher = require('lighthouse/chrome-launcher/chrome-launcher');
```

**Comment by paulirish** [comment]
fixed by #2262 and #2260
